contracts/TrueCurrencies/TrueRewardBackedToken.sol

```diff
@@ -229,95 +229,63 @@ contract TrueRewardBackedToken is RewardTokenWithReserve {
         return _toToken(opportunityRewardSupply(), opportunity());
     }
 
-    function redeemWithReserve(address sender, uint256 depositAmount, uint256 rewardAmount, address finOp) internal returns (uint256) {
-        if (reserveBalance() >= depositAmount) {
-            swapRewardForToken(sender, depositAmount, rewardAmount, finOp);
-            return depositAmount;
-        } else {
-            return redeemRewardToken(sender, rewardAmount, finOp);
-        }
-    }
-
-    function redeemFromSender(address sender, uint256 depositAmount) internal returns (uint256) {
-        if (!trueRewardEnabled(sender)) {
-            return depositAmount;
+    /**
+     * @dev Transfer helper function for TrueRewardBackedToken
+     */
+    function _transferAllArgs(address _from, address _to, uint256 _value) internal returns (address) {
+        if ((!trueRewardEnabled(_from) && !trueRewardEnabled(_to)) || _from == opportunity() || _to == opportunity()) {
```

Comment:
Come on folks, that suck soooo much.... previous version was much more readable. I'm in favour of using some 214 gas per request to make save devs minds.

Reply:
Totally agree